build.sh

```diff
@@ -14,3 +14,4 @@ done
 
 scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
 "$scriptroot/eng/common/build.sh" --projects "$scriptroot/illink.sln" --build --restore $@
+"$scriptroot/eng/dotnet.sh" tool run dotnet-format --check --verbosity diagnostic -f . --exclude src/analyzer,src/tuner,external
```

Comment:
I think this will still run the format even if there are build issues which is confusing. would be good to only do that if the previous command succeeded.

Reply:
we should probably also add an option to not do `--check` so it actually fixes the issues so that I don't need to manually apply suggestions.

Reply:
I'd suggest adding a new scripts `lint.sh` and `lint.cmd` which do the formating and call .sh one from CI script with additional `--check` argument